hyperspy/signal.py

```diff
@@ -1712,6 +1712,88 @@ def data(self, value):
         else:
             self._data = np.atleast_1d(np.asanyarray(value))
 
+    @property
+    def real(self):
+        """Get the real part of the data. Returns an :class:`~hyperspy.signals.Signal2D`."""
+        sig = self._deepcopy_with_new_data(np.real(self.data))
+        if sig.metadata.General.title:
+            title = sig.metadata.General.title
+        else:
+            title = 'Untitled Signal'
+        sig.metadata.General.title = 'real({})'.format(title)
+        return sig
+
+    @property
+    def imag(self):
+        """Get imaginary part of the data. Returns an :class:`~hyperspy.signals.Signal2D`."""
+        sig = self._deepcopy_with_new_data(np.imag(self.data))
+        if sig.metadata.General.title:
+            title = sig.metadata.General.title
+        else:
+            title = 'Untitled Signal'
+        sig.metadata.General.title = 'imag({})'.format(title)
+        return sig
+
+    def angle(self, deg=False):
+        """Return the angle of the (complex) data of the signal. Is zero if data is real.Calculate
+        the complex anglet the phase of the data.
```

Comment:
I think that the docstring could be more accurate. The angle is 0 for positive values and `2 * pi ` for negative values when data is real.

What about

    Return the angle (also known as phase or argument). If the data is real, the angle is 0 for positive values and 2pi for negative values.

?